packages/api/src/controllers/helpers.js

```diff
@@ -185,18 +193,22 @@ export async function trackAction(userId, email, event, apiKey) {
 }
 
 export async function fetchSegmentApi(body, endpoint, apiKey) {
+  const labels = { endpoint };
+  const timer = segmentMetrics.startTimer(labels);
   const segmentApiUrl = "https://api.segment.io/v1";
 
   const headers = {
     "Content-Type": "application/json",
     Authorization: "Basic " + Buffer.from(`${apiKey}:`).toString("base64"),
   };
 
-  await fetch(`${segmentApiUrl}/${endpoint}`, {
+  let response = await fetch(`${segmentApiUrl}/${endpoint}`, {
     method: "POST",
     body: JSON.stringify(body),
     headers: headers,
   });
+  labels.status_code = response.status.toString();
```

Comment:
Is there a way to send the labels only when we finish the timer, instead of modifying an object that has already been sent as an argument? I find that API kind of weird, encouraging mutable objects, surprising that it works.

Reply:
labels can be set anywhere. i was also surprised by mutable object observers, but it felt more js/ts. can be removed from here and be sent as param in the end

Reply:
Yeah I think that would be a better pattern IMO. Just a reminder that you changed on the prometheus client but not here.